File: 2021/2/1.py
x = 0
y = 0

FORWARD = "forward"
DOWN = "down"
UP = "up"

with open("input.txt") as f:
    for line in f:
        [direction, str_units] = line.split(" ")
        units = int(str_units)
        if direction == FORWARD:
            x += units
        elif direction == DOWN:
            y += units
        elif direction == UP:
            y -= units
        print(x, y)

print(x * y)

File: 2022/2/1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	f, _ := os.Open("input.txt")

	defer f.Close()

	scanner := bufio.NewScanner(f)
	totalScore := 0
	scoreMap := map[string]int{"X": 1, "Y": 2, "Z": 3}

	for scanner.Scan() {
		var resultScore int
		round := scanner.Text()

		// A, X = Rock
		// B, Y = Paper
		// C, Z = Scissors
		if round == "A Y" || round == "B Z" || round == "C X" {
			resultScore = 6
		} else if round == "A X" || round == "B Y" || round == "C Z" {
			resultScore = 3
		} else if round == "A Z" || round == "B X" || round == "C Y" {
			resultScore = 0
		}

		shapes := strings.Split(round, " ")
		shape2 := shapes[1]

		shapeScore := scoreMap[shape2]

		totalScore += shapeScore + resultScore
	}
	fmt.Printf("Total score is: %d", totalScore)
}
